Models/base_object.rb

```diff
@@ -0,0 +1,24 @@
+require 'securerandom'
+
+class BaseObject
+  attr_accessor :guid
+
+  def initialize(guid: nil)
+    (guid == nil) ? @guid = SecureRandom.uuid : @guid = guid
+  end
+
+  def get_as_delimiter_str(delimiter = ';')
+    result = ''
+    instance_variables.map do |attribute|
+    result += instance_variable_get(attribute).to_s + delimiter
```

Comment:
map looks like each here, better is 
result = instance_variables.map do |attribute| 
instance_variable_get(attribute).to_s + delimiter
end.join('  ')